podpac/core/coordinates/utils.py

```diff
@@ -539,6 +539,46 @@ def lower_precision_time_bounds(my_bounds, other_bounds, outer):
     return my_bounds, other_bounds
 
 
+def higher_precision_time_bounds(my_bounds, other_bounds, outer):
+    """
+    When given two bounds of np.datetime64, this function will convert both bounds to the higher-precision (in terms of 
+    time unit) numpy datetime4 object if outer==True, otherwise only my_bounds will be converted.
+    
+    Parameters
+    -----------
+    my_bounds : List(np.datetime64)
+        The bounds of the coordinates of the dataset
+    other_bounds : List(np.datetime64)
+        The bounds used for the selection
+    outer : bool
+        When the other_bounds are higher precision than the input_bounds, only convert these IF outer=True
+        
+    Returns
+    --------
+    my_bounds : List(np.datetime64)
+        The bounds of the coordinates of the dataset at the new precision
+    other_bounds : List(np.datetime64)
+        The bounds used for the selection at the new precision, if outer == True, otherwise return original coordinates    
+        
+    Notes
+    ------
+    When converting a YYYY-MM-DD to YYYY-MM-DD HH with outer=True, the largest value for HH is used, since the whole day is valid
+    """
+    if not isinstance(other_bounds[0], np.datetime64) or not isinstance(other_bounds[1], np.datetime64):
+        raise TypeError("Input bounds should be of type np.datetime64 when selecting data from:", str(my_bounds))
+
+    if not isinstance(my_bounds[0], np.datetime64) or not isinstance(my_bounds[1], np.datetime64):
+        raise TypeError("Native bounds should be of type np.datetime64 when selecting data using:", str(other_bounds))
+
+    if my_bounds[0].dtype > other_bounds[0].dtype and outer:
+        sign = [0, 1]
+        other_bounds = [(b + s).astype(my_bounds[0].dtype) - 1 for s, b in zip(sign, other_bounds)]
```

Comment:
Not completely sure this line is as it should be... in particular perhaps it should be 
```suggestion
        other_bounds = [(b + s).astype(my_bounds[0].dtype) - s for s, b in zip(sign, other_bounds)]
```

**note** `-1` change to `s`

Reply:
Yeah, it definitely *seems* like it should be  ` - s` instead of `- 1`.

I found this easier to think about unpacked:

```
other_bounds = [
    (other_bounds[0] + 0).astype(my_bounds[0].dtype) - 0,
    (other_bounds[1] + 1).astype(my_bounds[0].dtype) - 1
]
```

or probably just:

```
other_bounds = [
    other_bounds[0].astype(my_bounds[0].dtype),
    (other_bounds[1] + 1).astype(my_bounds[0].dtype) - 1
]
```

 Can you explain what the `+1 -1` does? We should add a comment about that.

Reply:
Okay, I get it the `+1` and `-1`. I'll add a comment.